docs/api/vega-wallet/reference/browser-extension/javascript-api.md

```diff
@@ -0,0 +1,260 @@
+---
+title: JavaScript API
+hide_title: false
+vega_network: TESTNET
+sidebar_position: 2
+---
+import Topic from '/docs/topics/_topic-wallet.mdx'
+
+<Topic />
+
+## Vega Browser Wallet
+
+dApps and websites can communicate with the Vega browser wallet over 
+JSON-RPC 2.0, however the browser extension also provides
+a high-level Javascript API, injected into web pages.
+
+### Detecting the extension
+
+The Vega browser wallet injects methods on the global `vega` object,
+hence the presence of the extension can be detected by checking that
+this object is defined:
+
+```js
+if (globalThis.vega != null) {
```

Comment:
The eth docs seem to use window, i guess we should be using this as lowest common denominator if possible.

Reply:
Yeah good call. I used `globalThis` as that's `window` but also works in WebWorkers and ServiceWorkers, but I don't know if extensions are exposed in such contexts